src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Import.cs

```diff
@@ -35,9 +47,70 @@ public static CngKey Import(byte[] keyBlob, CngKeyBlobFormat format, CngProvider
 
             SafeNCryptProviderHandle providerHandle = provider.OpenStorageProvider();
             SafeNCryptKeyHandle keyHandle;
-            ErrorCode errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            ErrorCode errorCode;
+            
+            if (curve == null)
+            {
+                errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
+                if (errorCode != ErrorCode.ERROR_SUCCESS)
+                {
+                    Exception e = errorCode.ToCryptographicException();
+                    if (errorCode == ErrorCode.NTE_NOT_SUPPORTED &&
+                        (format == CngKeyBlobFormat.EccFullPrivateBlob || format == CngKeyBlobFormat.EccFullPrivateBlob))
+                    {
+                        // Attempt to import on < Windows 10
+                        throw new PlatformNotSupportedException(SR.Cryptography_InvalidCurveOid, e);
```

Comment:
That feels like a bad fit to me.  Plus, it could be something like someone trying to import a key into a smartcard that doesn't support it.

errorCode.ToCryptographicException() seems the most straightforward, if uninformative.  There're just too many reasons it could've gone wrong in the underlying provider.